eng/versioning/version_shared.py

```diff
@@ -24,34 +38,39 @@ def is_metapackage(package_path):
     azure_path = path.join(dir_path, 'azure')
     return not path.exists(azure_path)
 
-def get_setup_py_paths(base_path):
-    glob_expression = path.join(base_path, SETUP_PY_GLOB)
-    setup_paths = glob(glob_expression, recursive=True)
-
-    filtered_paths = [path for path in setup_paths if not path_excluded(path)]
+def get_setup_py_paths(glob_string, base_path):
+    setup_paths = process_glob_string(glob_string, base_path)
+    filtered_paths = [path.join(p, 'setup.py') for p in setup_paths if not path_excluded(p)]
     return filtered_paths
 
 
-def get_packages(sdk_location):
-    paths = get_setup_py_paths(sdk_location)
+def get_packages(args):
```

Comment:
I would rather that we do _not_ pass in the args object here. IMO this will make us need to glance through the code to find out how to name the parameter in the `args` object. Given that we only look for `args.service`, I'd prefer that we just pass in the service directory directly.

Reply:
Updated comment in code to mention the list of arguments processed by shared function getpackages()